### ISSUE TITLE
"Wait for dids and publish to mqtt..."

### ISSUE THREAD

**Issue by Donboxo** [MyHomeMyData/E3onCAN#4]
Hi,

I have two issues:
#1 paho_mqtt has been updated to V2 and this is why it does not publish to mqtt anymore

I can downgrade paho with
pip install "paho-mqtt<2.0.0"
But when i do this I get issue
#2 
I want to publish to MQTT with the following:
python3 E3onCANcollect.py -c can0 -dev e380 -r 592 -m localhost:1883:open3e -retain 592
When I do this I get the following message:
_Wait for dids and publish to mqtt..._
with no publishing to my MQTT broker

**Comment by MyHomeMyData** [comment]
Hi @Donboxo,
regarding #1:
* I used the same command line and it's working fine with paho-mqtt V1.6.1
* pls. check if the E380 data is present on bus: `candump can0 -tA | grep 250`
* pls. check if your mqtt broker is listening to the right topics. You may use mosquitto: `mosquitto_sub -v -t open3e/#`

regarding #2:
* Thank you for the hint. I will update E3onCAN to paho-mqtt V2

Btw.: Did you know, there is an adpater for ioBroker available supporting E380 data collection? You my install from [here](https://github.com/MyHomeMyData/ioBroker.e3oncan) or via Beta-Repository in ioBroker (search for e3oncan or viessmann).

**Comment by Donboxo** [comment]
Thanks so much!
With 1.6.1 it works now. I also forgot to include mqtt password...
Is there a way to increase the interval readout? I think it is at 250 ms at the moment and I want to set it to 1 s

I currently use the E380 readout only for including the Grid Power into evcc for my electric car, but in the future I will use ioBroker for data collection so thanks for the hint!

**Comment by MyHomeMyData** [comment]
@Donboxo  Good to here it's working. I just pushed the update for paho-mqtt V2. Pls. run `pip install -r requirements.txt` after updating.

> Is there a way to increase the interval readout?

That's not possible with present implementation. All incoming data is just forwarded to mqtt. If it's important four you, pls. place an issue. ioBroker adapter can do such limitation already.

Closing this issue.
